### ISSUE TITLE
Yolov3 doesn't detect classes on test images

### ISSUE THREAD

**Issue by kingsley351** [AlexeyAB/darknet#5098]
I have trained yolo v3 on 4 classes in text document images (latitude, longitude, X, Y).
The loss function decreases rapidly in 1000 iterations towards the average of 0.964344

995: 1.007693, 1.031876, 0.000980 , 12.066782, 63680
996: 1.076354, 1.036324, 0.000984 , 12.603208, 63744
997: 0.801689, 1.012860, 0.000988 , 12.560430, 63808
998: 0.925991, 1.004173, 0.000992 , 12.780347, 63872
999: 0.930889, 0.996845, 0.000996 , 13.021864, 63936
1000: 0.671834, 0.964344, 0.001000 , 12.081135, 64000
1001: 0.796824, 0.947592, 0.001000 , 12.564254, 64064
1002: 0.914025, 0.944235, 0.001000 , 12.430697, 64128
1003: 0.674278, 0.917240, 0.001000 , 11.264583, 64192
1004: 0.707057, 0.896221, 0.001000 , 12.001313, 64256
1005: 0.673263, 0.873926, 0.001000 , 12.418809, 64320
1006: 0.644836, 0.851017, 0.001000 , 12.574316, 64384
1007: 0.695713, 0.835486, 0.001000 , 11.431092, 64448
1008: 0.731694, 0.825107, 0.001000 , 11.592270, 64512
1009: 0.738076, 0.816404, 0.001000 , 11.477111, 64576
1010: 0.901874, 0.824951, 0.001000 , 11.840801, 64640
1011: 0.777433, 0.820199, 0.001000 , 9.353921, 64704
1012: 0.664819, 0.804661, 0.001000 , 9.603042, 64768
1013: 0.749343, 0.799129, 0.001000 , 9.857964, 64832
1014: 0.633777, 0.782594, 0.001000 , 9.195294, 64896
1015: 0.607779, 0.765112, 0.001000 , 9.469180, 64960

When I made the test on some training set images, the detection was accurate but when I test other images, Yolo didn't detect nothing.

The positions of the classes in the other images of test are different to training set. But normaly yolo convolutionnal layers should detect these classes because latitude & longitude are kind of regular expressions.

Could you please help me ?

**Comment by kingsley351** [comment]
these are the results of map function after 2200 iterations, avg_loss = 0.306350

detections_count = 366, unique_truth_count = 130  
class_id = 0, name = lat, ap = 40.26%   	 (TP = 16, FP = 22) 
class_id = 1, name = long, ap = 24.36%   	 (TP = 11, FP = 19) 
class_id = 2, name = Y, ap = 54.29%   	 (TP = 21, FP = 11) 
class_id = 3, name = X, ap = 37.69%   	 (TP = 18, FP = 13) 

 for conf_thresh = 0.25, precision = 0.50, recall = 0.51, F1-score = 0.51 
 for conf_thresh = 0.25, TP = 66, FP = 65, FN = 64, average IoU = 35.02 % 

 IoU threshold = 50 %, used Area-Under-Curve for each unique Recall 
 mean average precision (mAP@0.50) = 0.391484, or 39.15 % 
Total Detection Time: 23 Seconds

**Comment by AlexeyAB** [comment]
Show examples of detections

**Comment by kingsley351** [comment]
![test-1](https://user-images.githubusercontent.com/62565568/77379270-95813f00-6d78-11ea-8eb9-15e70c142da8.jpg)

**Comment by kingsley351** [comment]
![test-3](https://user-images.githubusercontent.com/62565568/77379293-a2059780-6d78-11ea-8ad3-610bfa9bc35d.jpg)

**Comment by kingsley351** [comment]
The first image is the test of one image from the training set, the second one is new image.
Yolo doesn't detect nothing in the second image who it never seen before.

**Comment by AlexeyAB** [comment]
https://github.com/AlexeyAB/darknet#how-to-improve-object-detection

> for each object which you want to detect - there must be at least 1 similar object in the Training dataset with about the same: shape, side of object, **relative size,** angle of rotation, tilt, illumination. So desirable that your training dataset include images with objects at diffrent: scales, rotations, lightings, from different sides, on different backgrounds - you should preferably have 2000 different images for each class or more, and you should train 2000*classes iterations or more

**Comment by kingsley351** [comment]
The MAP function did not exceed 56%, Is that normal ?
